### PR TITLE
Epoll fixes

### PR DESCRIPTION
**Pull request by peters**
## What does the pull request do?
Throws a managed exception if a native method call fail.

## What is the current behavior?
Returned error codes are not handled properly.

## What is the updated/expected behavior with this PR?
I'm investigating a hard crash on ARM64. It might be CLR related but I need to make sure Avalonia
is handling errors correctly.

## Checklist

- [ ] Added unit tests (if possible)?
- [ ] Added XML documentation to any related classes?
- [ ] Consider submitting a PR to https://github.com/AvaloniaUI/Avaloniaui.net with user documentation

### CODE REVIEW

**Comment by dnfadmin** [comment]
[CLA assistant check](https://cla.dotnetfoundation.org/AvaloniaUI/Avalonia?pullRequest=5671) <br/>Thank you for your submission, we really appreciate it. Like many open source projects, we ask that you sign our [Contributor License Agreement](https://cla.dotnetfoundation.org/AvaloniaUI/Avalonia?pullRequest=5671) before we can accept your contribution.<br/><br/>:x: peters [sign now](https://cla.dotnetfoundation.org/AvaloniaUI/Avalonia?pullRequest=5671)<br/><sub>You have signed the CLA already but the status is still pending? Let us [recheck](https://cla.dotnetfoundation.org/check/AvaloniaUI/Avalonia?pullRequest=5671) it.</sub>

File: src/Avalonia.X11/X11PlatformThreading.cs [hunk 1]
```diff
@@ -235,10 +236,15 @@ public void RunLoop(CancellationToken cancellationToken)
...
                 //Flush whatever requests were made to XServer
                 XFlush(_display);
-                epoll_event ev;
                 if (XPending(_display) == 0)
-                    epoll_wait(_epoll, &ev, 1,
-                        nextTick == null ? -1 : Math.Max(1, (int)(nextTick.Value - _clock.Elapsed).TotalMilliseconds));
+                {
+                    var epollWaitTimeout = nextTick == null ? -1 : Math.Max(1, (int)(nextTick.Value - _clock.Elapsed).TotalMilliseconds);
+                    var epollWaitStatus = epoll_wait(_epoll, out _, 1, epollWaitTimeout);
+                    if (epollWaitStatus == -1)
```
Comment (peters) [inline src/Avalonia.X11/X11PlatformThreading.cs:243]: @kekekeks I guess we might not want to throw if `epoll_wait` was interrupted by a signal handler, e.g. `SIGABRT` ?

Comment (kekekeks) [reply src/Avalonia.X11/X11PlatformThreading.cs:243]: epoll_wait is used here only to save CPU cycles, so return value can be ignored.

**Comment by peters** [comment]
@kekekeks The crash is not related to Avalonia 🚀

**Pull request closed without merging**